src/js/components/devices/device-details/monitoring.js

```diff
@@ -4,15 +4,15 @@ import Time from 'react-time';
 import DeviceDataCollapse from './devicedatacollapse';
 import { DeviceOfflineHeaderNotification, NoAlertsHeaderNotification, severityMap } from './notifications';
 
-const MonitoringAlert = ({ alert: { id, level, name, subject, timestamp }, onLogClick }) => (
+const MonitoringAlert = ({ alert, onLogClick }) => (
   <div className="monitoring-alert column-data">
-    {severityMap[level].listIcon}
+    {severityMap[alert.level].listIcon}
     <div className="key text-muted">
-      <b>{name}</b>
+      <b>{alert.name}</b>
     </div>
-    <div>{level}</div>
-    <Time value={timestamp} format="YYYY-MM-DD HH:mm" />
-    {subject.details ? <a onClick={() => onLogClick(id, subject.details)}>view log</a> : null}
+    <div>{alert.level}</div>
+    <Time value={alert.timestamp} format="YYYY-MM-DD HH:mm" />
+    {Object.keys(alert.subject.details).some(key => key.startsWith('line')) && <a onClick={() => onLogClick(alert)}>view log</a>}
```

Comment:
this means that we are not showing "details" from the subject in any other case than when log lines are present. could we check `subject.status`? if it is `LOGCONTAINS` then we can expect to have lines before and after, if not we can show `subject.details.description`, if that is not present, then we show nothing.
The reason is, for crond not running, alert lokos like that:
```json
{
	"_id" : UUID("15be265a-bcf9-440c-87db-dd4cc3f1f14d"),
	"name" : "Service not running",
	"device_id" : "2f62a3b3-0f83-41f6-9903-e68d6f25958f",
	"level" : 128,
	"subject" : {
		"name" : "cron",
		"type" : "service",
		"status" : "not-running",
		"details" : {
			"description" : "The main process is not present"
		}
	},
	"timestamp" : ISODate("2021-09-03T20:40:27Z"),
	"exp" : ISODate("2021-10-03T19:40:28.092Z"),
	"tenant_id" : "5abcb6de7a673a0001287c71"
}
```
while for LOGCONTAINS case (pattern found in logs) it may look like that (no lines before or after available):
```json
{
	"_id" : UUID("e557b5b7-483e-4e2f-bd2d-b767110f037a"),
	"name" : "Log file contains this one",
	"device_id" : "2f62a3b3-0f83-41f6-9903-e68d6f25958f",
	"level" : 128,
	"subject" : {
		"name" : "tmp",
		"type" : "log",
		"status" : "LOGCONTAINS",
		"details" : {
			"description" : "this one present in /tmp/l.logfile"
		}
	},
	"timestamp" : ISODate("2021-09-03T06:00:08Z"),
	"exp" : ISODate("2021-10-03T05:00:09.051Z"),
	"tenant_id" : "5abcb6de7a673a0001287c71"
}
```
or like that, of lines before/atfer are available:
```json
{
	"_id" : UUID("ab051bb1-cb55-4809-a86b-c5cd1af0a277"),
	"name" : "Log file contains this one",
	"device_id" : "2f62a3b3-0f83-41f6-9903-e68d6f25958f",
	"level" : 128,
	"subject" : {
		"name" : "tmp",
		"type" : "log",
		"status" : "LOGCONTAINS",
		"details" : {
			"description" : "this one present in /tmp/l.logfile 1 times",
			"lines_before" : [
				{
					"line_number" : NumberLong(13),
					"data" : "12.ta12"
				},
				{
					"line_number" : NumberLong(14),
					"data" : "13.ta13"
				},
				{
					"line_number" : NumberLong(15),
					"data" : "14.ta14"
				},
				{
					"line_number" : NumberLong(16),
					"data" : "15.ta15"
				}
			],
			"lines_after" : [
				{
					"line_number" : NumberLong(18),
					"data" : "17.ta16"
				},
				{
					"line_number" : NumberLong(19),
					"data" : "18.ta17"
				},
				{
					"line_number" : NumberLong(20),
					"data" : "19.ta18"
				},
				{
					"line_number" : NumberLong(21),
					"data" : "20.ta19"
				}
			],
			"line_matching" : {
				"line_number" : NumberLong(16),
				"data" : "16.this one"
			}
		}
	},
	"timestamp" : ISODate("2021-09-01T17:58:06Z"),
	"exp" : ISODate("2021-10-01T16:58:06.654Z"),
	"tenant_id" : "5abcb6de7a673a0001287c71"
}
```
please note that lines before and/or after maybe empty, in which case we do not show the "expandable" controls before and/or after the line_matched